skills/repo.go

```diff
@@ -19,10 +19,75 @@
 
 package skills
 
+import (
+	"errors"
+	"sort"
+	"sync"
+)
+
 // Repository stores all known snappy skills and slots and types.
-type Repository struct{}
+type Repository struct {
+	// Protects the internals from concurrent access.
+	m     sync.Mutex
+	types []Type
```

Comment:
Please use a map here. This is the most obvious need we have for this data structure.